packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,427 @@
+  public readonly userAccounts$: Observable<ReadonlyArray<UserAccount>>;
+  public readonly networks$: Observable<ReadonlyArray<NetworkType>>;
+  private readonly dapi: Dapi;
+  private readonly initPromise: Promise<void>;
+  private readonly provider: TProvider;
+  private readonly currentAccountInternal$: BehaviorSubject<UserAccount | undefined>;
+  private readonly userAccountsInternal$: BehaviorSubject<ReadonlyArray<UserAccount>>;
+  private readonly networksInternal$: BehaviorSubject<ReadonlyArray<NetworkType>>;
+  private readonly defaultNetworkInternal$: BehaviorSubject<NetworkType | undefined>;
+
+  public constructor({ dapi, provider }: { readonly dapi: Dapi; readonly provider: TProvider }) {
+    super();
+    this.dapi = dapi;
+    this.provider = provider;
+    this.initPromise = this.init();
+
+    this.userAccountsInternal$ = new BehaviorSubject<ReadonlyArray<UserAccount>>([]);
+    this.currentAccountInternal$ = new BehaviorSubject<UserAccount | undefined>(undefined);
+    this.networksInternal$ = new BehaviorSubject<ReadonlyArray<NetworkType>>([]);
+    this.defaultNetworkInternal$ = new BehaviorSubject<NetworkType | undefined>(undefined);
+
+    this.dapi.addEventListener('ACCOUNT_CHANGED', ({ address, label }) => {
+      this.dapi
+        .getPublicKey()
+        .then(({ publicKey }) => {
+          this.updateUserAccount({
+            network: this.defaultNetworkInternal$.getValue() as NetworkType,
+            publicKey,
+            address,
+            label,
+          });
+        })
+        .catch(() => {
+          // do nothing
+        });
+    });
+    this.dapi.addEventListener('NETWORK_CHANGED', (dapiNetworks) => {
+      this.updateNetworks(dapiNetworks);
+    });
+    this.dapi.addEventListener('DISCONNECTED', () => {
+      throw new Error(`The account connected to the dapp via the dapi provider has been disconnected (logged out).`);
+    });
+
+    this.userAccounts$ = this.userAccountsInternal$;
+    this.currentUserAccount$ = this.currentAccountInternal$.pipe(distinctUntilChanged());
+    this.networks$ = this.networksInternal$;
+  }
+
+  public getCurrentUserAccount(): UserAccount | undefined {
+    return this.currentAccountInternal$.getValue();
+  }
+
+  public async selectUserAccount(_userAccountID?: UserAccountID): Promise<void> {
+    throw new NotImplementedError('selectUserAccount');
+  }
+
+  public async deleteUserAccount(_userAccountID?: UserAccountID): Promise<void> {
+    throw new NotImplementedError('deleteUserAccount');
+  }
+
+  public getUserAccounts(): ReadonlyArray<UserAccount> {
+    return this.userAccountsInternal$.getValue();
+  }
+
+  public getNetworks(): ReadonlyArray<NetworkType> {
+    return this.networksInternal$.getValue();
+  }
+
+  public async updateUserAccountName(_options: UpdateAccountNameOptions): Promise<void> {
+    throw new NotImplementedError('updateUserAccountName');
+  }
+
+  public async getBlockCount(network: NetworkType): Promise<number> {
```

Comment:
This can be moved to the base class, just add `provider` as a parameter to the constructor.